rust/otap-dataflow/crates/engine/src/local/processor.rs

```diff
@@ -205,6 +214,30 @@ impl<PData> EffectHandler<PData> {
         self.core.start_periodic_timer(duration).await
     }
 
+    /// Reply in failure
+    pub async fn reply_nack(
+        &self,
+        node_id: Option<usize>,
```

Comment:
I probably don't understand something here. Why `node_id` here and why is it an option?